hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/BucketIdentifier.java

```diff
@@ -54,15 +54,30 @@ private static List<String> getHashKeysUsingIndexFields(String recordKey, List<S
   }
 
   public static String partitionBucketIdStr(String partition, int bucketId) {
-    return String.format("%s_%s", partition, bucketIdStr(bucketId));
```

Comment:
Does it make sense we add a cache for it, just like what we do to `avro schema string -> schema` conversion.